web/src/components/upload/index.js

```diff
@@ -1,25 +1,32 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import moment from "moment-timezone";
 
 import { UploadDrop } from "./drop";
 import { UploadConfirm } from "./confirm";
 import { Container } from "../ui";
+import { getDevices } from "./drop.utils";
 
 export const Upload = () => {
   const [files, setFiles] = useState([]);
   const [phase, setPhase] = useState("drop"); //"drop" or "confirm" page
   const [dustrakStart, setDustrakStart] = useState(moment(""));
   const [dustrakEnd, setDustrakEnd] = useState(moment(""));
   const [dustrakSerial, setDustrakSerial] = useState("");
+  const [devices, setDevices] = useState([]);
 
-  /**
-   * Map of each dustrak serial number to its label
-   */
-  const dustrakSerialToLabel = {
-    8530091203: "Device A",
-    8530094612: "Device B",
-    8530100707: "Device C",
-  };
+  useEffect(() => {
+    (async () => {
+      try {
+        const deviceList = await getDevices();
+        setDevices(deviceList);
+      } catch(err) {
+        console.error(err);
+      }
+    })()
+  }, [dustrakSerial]);
+
+  const matchedDevice = devices.filter(device => device.serial === dustrakSerial);
+  const deviceName = matchedDevice.length ? matchedDevice[0].name : null;
```

Comment:
1. As we expect the device to change when the dustrakSerial changes, we should track it with state.
2. As we iterate on the design, we'll want to save the device association in the database. In which case, we'll want the device id, too. We might as well keep the whole device object and pass it to the confirm component.
3. If we set the 'error' value to an empty version of its type, rather than 'null', then we can more reliably and concisely handle the error paths. With the change to using the device object, we would default to using an empty object.